pkg/cosign/kubernetes/webhook/validator.go

```diff
@@ -153,6 +166,26 @@ func (v *Validator) validatePodSpec(ctx context.Context, ps *corev1.PodSpec, opt
 	return errs
 }
 
+func getAuthorityKeys(ctx context.Context, ref name.Reference, config *config.Config) (keys []*ecdsa.PublicKey, errs *apis.FieldError) {
```

Comment:
I'm not sure we want to make these as apis.FieldError. We usually use those as validation time errors because they are user facing directly. I'd just been thinking that we'll use err for the return value, not apis.FieldError. <time passes...> But then as I started thinking about it some more, the are user facing at least in some cases because they will be returned to the user when they try to deploy something.

Reply:
I also felt similar that it should have been `errors` initially. I followed using `apis.ErrGeneric` because `getAuthorityKeys/parseAuthorityKeys` followed the same logic from `getKeys`. `getKeys` returned errors using `apis.ErrGeneric`. I just tried to surface the lowest level error up and made it a common type. Should I change it such that getAuthorityKeys returns errors only and then have `validatePodSpec` determine how to interpret generic `error` such as into apis.FieldError? 

Or is this something that can be changed later when we actually start erroring out more concretely? The change seems trivial enough either way you want it.

Reply:
Question: why do we even need to parse the authority keys here ? The function call comes from validatePodSpec, i assume the parsing of the authorities has been done when creating the image policy config, rather when validating the pod spec.

Reply:
When we get the `ImagePolicyConfig`, it contains `Policies.[]ClusterImagePolicySpec.[]Authority`. The returned `Authority` contains `KeyRef.Data` which is a string. `parseAuthorityKeys` takes that string and parses it into `ecdsa.PublicKey`.

Reply:
Well, sorry I was thinking that we could be defining the signature as error, I think at least (too lazy to look up as I'm on the phone) we could still be returning apierrors as they are errors but not bleed that into the signature of the function.